### ISSUE TITLE
`flake8-type-checking`: enforce combining imports from inside and outside a `TYPE_CHECKING` block (reverse of `strict`)

### ISSUE THREAD

**Issue by GideonBear** [astral-sh/ruff#21921]
### Summary

When using `tool.ruff.lint.flake8-type-checking.strict = false` (the default), ruff will not split up imports into a `TYPE_CHECKING` block:
```py
from __future__ import annotations

from m import A, B

a = A()
b: B
```
```console
$ \cat pyproject.toml
tool.ruff.lint.flake8-type-checking.strict = false
$ ruff check main.py --select TC
All checks passed!
$ nano pyproject.toml 
$ \cat pyproject.toml
tool.ruff.lint.flake8-type-checking.strict = true
$ ruff check main.py --select TC
TC002 Move third-party import `m.B` into a type-checking block
etc.
```

`strict = true` is great for consistency. However when using `strict = false`, there is no way to enforce having these imports combined, making it impossible to enforce consistency.

Current behavior:
```py
from __future__ import annotations

from typing import TYPE_CHECKING

from m import A

if TYPE_CHECKING:
    from m import B

a = A()
b: B
```
```console
$ ruff check main.py --select TC
All checks passed!
```
Desired behavior: suggest combining the imports (to what it was in the first example).

**Comment by GideonBear** [comment]
Soft ping to @Daverball, thoughts?

**Comment by ntBre** [comment]
If I understand correctly, this is a duplicate of https://github.com/astral-sh/ruff/issues/19548.

**Comment by Daverball** [comment]
Looks like a duplicate to me as well. Overall I don't dislike the idea, but it definitely can't be a boolean setting at this point. We don't want to cause arguably unnecessary churn in either direction, unless people specifically opt into it for aesthetics/consistency. So it would have to be a three-stage setting, either you prefer imports staying together, you prefer all the typing only imports staying in type checking blocks, even if it means splitting up imports or you have no preference and ruff  will only nag you when there's a tangible benefit to moving the import beyond aesthetics/consistency.

**Comment by GideonBear** [comment]
> If I understand correctly, this is a duplicate of [#19548](https://github.com/astral-sh/ruff/issues/19548).

It is, looks like I missed that. Thanks!

**Comment by GideonBear** [comment]
(accidentally closed as completed, meant to close as duplicate)

**Comment by vivodi** [comment]
> (accidentally closed as completed, meant to close as duplicate)

You can still reopen and close it as duplicate.

**Comment by GideonBear** [comment]
> > (accidentally closed as completed, meant to close as duplicate)
> 
> You can still reopen and close it as duplicate.

Huh, the button *was* greyed-out for me, I assumed I didn't have permission to do that. Thx
